internal/services/costmanagement/scheduled_action_resource_base.go

```diff
@@ -0,0 +1,374 @@
+			Required:     true,
+			ValidateFunc: validation.IsRFC3339Time,
+		},
+
+		"end_date": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.IsRFC3339Time,
+		},
+	}
+
+	for k, v := range fields {
+		output[k] = v
+	}
+
+	return output
+}
+
+func (br costManagementScheduledActionBaseResource) attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (br costManagementScheduledActionBaseResource) createFunc(resourceName, scopeFieldName string) sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.CostManagement.ScheduledActionsClient_v2022_10_01
+			id := scheduledactions.NewScopedScheduledActionID(metadata.ResourceData.Get(scopeFieldName).(string), metadata.ResourceData.Get("name").(string))
+
+			existing, err := client.GetByScope(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+				}
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return tf.ImportAsExistsError(resourceName, id.ID())
+			}
+
+			emailAddressesRaw := metadata.ResourceData.Get("email_addresses").(*pluginsdk.Set).List()
+			emailAddresses := utils.ExpandStringSlice(emailAddressesRaw)
+
+			viewId := views.NewScopedViewID(metadata.ResourceData.Get(scopeFieldName).(string), metadata.ResourceData.Get("view_name").(string))
+
+			var daysOfWeek []scheduledactions.DaysOfWeek
+			if metadata.ResourceData.Get("days_of_week").(*pluginsdk.Set).Len() > 0 {
+				daysOfWeek = make([]scheduledactions.DaysOfWeek, 0)
+				for _, value := range metadata.ResourceData.Get("days_of_week").(*pluginsdk.Set).List() {
+					daysOfWeek = append(daysOfWeek, scheduledactions.DaysOfWeek(value.(string)))
+				}
+			}
+
+			var weeksOfMonth []scheduledactions.WeeksOfMonth
+			if metadata.ResourceData.Get("weeks_of_month").(*pluginsdk.Set).Len() > 0 {
+				weeksOfMonth = make([]scheduledactions.WeeksOfMonth, 0)
+				for _, value := range metadata.ResourceData.Get("weeks_of_month").(*pluginsdk.Set).List() {
+					weeksOfMonth = append(weeksOfMonth, scheduledactions.WeeksOfMonth(value.(string)))
+				}
+			}
+			schedule := scheduledactions.ScheduleProperties{
+				Frequency:    scheduledactions.ScheduleFrequency(metadata.ResourceData.Get("frequency").(string)),
+				WeeksOfMonth: &weeksOfMonth,
+				DaysOfWeek:   &daysOfWeek,
+				HourOfDay:    utils.Int64(int64(metadata.ResourceData.Get("hour_of_day").(int))),
+				DayOfMonth:   utils.Int64(int64(metadata.ResourceData.Get("day_of_month").(int))),
+				StartDate:    metadata.ResourceData.Get("start_date").(string),
+				EndDate:      metadata.ResourceData.Get("end_date").(string),
+			}
+
+			props := scheduledactions.ScheduledAction{
+				Kind: utils.ToPtr(scheduledactions.ScheduledActionKindEmail),
```

Comment:
The same API is also used to create `azurerm_cost_anomaly_alert` (`Kind=Insight Alert`), you could argue that this one is a `azurerm_subscription_scheduled_cost_email` or `azurerm_subscription_cost_management_scheduled_view_email`.., the question here is what the scope is of this resource and what Microsoft seems fit long term. It might be that this resource is extended to have more `Kind`s, or that it might extend to email multiple resource types: next to views also CSV files based on a query, whatsoever.

In the API docs they are also talking about `reports`, in the [Cost Management Docs](https://learn.microsoft.com/en-us/azure/cost-management-billing/costs/save-share-views#subscribe-to-scheduled-alerts) it is referred to as `scheduled_alerts`..

Options:
- [ ] `azurerm_subscription_cost_management_scheduled_view_email` (Most precise technical description)
- [ ] `azurerm_subscription_cost_management_scheduled_report` (Best functional description IMHO)
- [ ] `azurerm_subscription_cost_management_scheduled_alert` (Closest to docs, confusing to me tbh)

Your choice 😵